### DISCUSSION TITLE
Backref relationships don't populate in the class until instance is created

### DISCUSSION THREAD

**Discussion by jinschoi** [sqlalchemy/sqlalchemy#7313] [Usage Questions] [answered]
### Describe the bug

Referencing a backref relationship raises an AttributeError unless an instance is instantiated first.
 
This causes a problem when the first thing you do is try to use a 2.0-style select() query using one of the relationships in a condition. It is easy enough to workaround, but the relationship should exist at the start.

### To Reproduce

```python
from sqlalchemy import *
from sqlalchemy.orm import *

Base = declarative_base()

class Parent(Base):
    __tablename__ = 'parent'
    id = Column(Integer, primary_key=True)
    children = relationship('Child', backref='parent')

class Child(Base):
    __tablename__ = 'child'
    id = Column(Integer, primary_key=True)
    parentid = Column(Integer, ForeignKey('parent.id'))

# Uncommenting this makes the relationship available.
# c = Child()

Child.parent
```

### Error

```
>>> Traceback (most recent call last):
>>> Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File ".../test.py", line 19, in <module>
    Child.parent
AttributeError: type object 'Child' has no attribute 'parent'
``` 

### Versions

- OS: macOS 12.0.1
- Python: 3.8.12
- SQLAlchemy: 1.4.23
- Database: all
- DBAPI (eg: psycopg, cx_oracle, mysqlclient): all

**Comment by CaselIT** [comment, marked as answer]
Hi,

This seems due to the lazy initialization of the modules. You should place a call to `configure_mappers()` one they are all imported. I don't think there is much we can do without breaking other use cases here.

The suggested alternative is to avoid backref and use instead backpopulate with an explicit relationship.

**Reply by jinschoi** [reply]
Thanks. I was looking for something like configure_mappers(), but didn't know what I should be searching for.

**Reply by CaselIT** [reply]
also consider [`back_populates`](https://docs.sqlalchemy.org/en/14/orm/relationship_api.html#sqlalchemy.orm.relationship.params.back_populates) to make thinks less magic but usually cleared to reason about
